kernel-open/nvidia/nv-pci.c

```diff
@@ -434,6 +506,12 @@ nv_pci_probe
         goto failed;
     }
 
+    if (nv_resize_pcie_bars(pci_dev)) {
```

Comment:
Since this path runs every time .probe is called, we should add this feature behind a regkey module param at first (see nvrm_registry.h). Once we have good test coverage and feel confident that the feature is robust, we can enable this feature by default.

Reply:
Do you mean `nv-reg.h`? Those expose kernel module options?